Sources/StreamChatSwiftUI/ChatThreadList/ChatThreadList.swift

```diff
@@ -0,0 +1,123 @@
+    
+    private var onItemTap: (ChatThread) -> Void
+    private var onItemAppear: (Int) -> Void
+
+    @ViewBuilder
+    private var headerView: () -> HeaderView
+
+    @ViewBuilder
+    private var footerView: () -> FooterView
+
+    public init(
+        factory: Factory,
+        threads: LazyCachedMapCollection<ChatThread>,
+        threadDestination: @escaping (ChatThread) -> Factory.ThreadDestination,
+        selectedThread: Binding<ThreadSelectionInfo?>,
+        onItemTap: @escaping (ChatThread) -> Void,
+        onItemAppear: @escaping (Int) -> Void,
+        headerView: @escaping () -> HeaderView,
+        footerView: @escaping () -> FooterView
+    ) {
+        self.factory = factory
+        self.threads = threads
+        self.threadDestination = threadDestination
+        self._selectedThread = selectedThread
+        self.onItemTap = onItemTap
+        self.onItemAppear = onItemAppear
+        self.headerView = headerView
+        self.footerView = footerView
+    }
+
+    public var body: some View {
+        ScrollView {
+            headerView()
+            ThreadsLazyVStack(
+                factory: factory,
+                threads: threads,
+                threadDestination: threadDestination,
+                selectedThread: $selectedThread, 
+                onItemTap: onItemTap,
+                onItemAppear: onItemAppear
+            )
+            footerView()
+        }
+    }
+}
+
+/// LazyVStack displaying list of threads.
+public struct ThreadsLazyVStack<Factory: ViewFactory>: View {
+    @Injected(\.colors) private var colors
+
+    private var factory: Factory
+    var threads: LazyCachedMapCollection<ChatThread>
+    private var threadDestination: (ChatThread) -> Factory.ThreadDestination
+    @Binding private var selectedThread: ThreadSelectionInfo?
+    private var onItemTap: (ChatThread) -> Void
+    private var onItemAppear: (Int) -> Void
+
+    public init(
+        factory: Factory,
+        threads: LazyCachedMapCollection<ChatThread>,
+        threadDestination: @escaping (ChatThread) -> Factory.ThreadDestination,
+        selectedThread: Binding<ThreadSelectionInfo?>,
+        onItemTap: @escaping (ChatThread) -> Void,
+        onItemAppear: @escaping (Int) -> Void
+    ) {
+        self.factory = factory
+        self.threads = threads
+        self.threadDestination = threadDestination
+        self.onItemTap = onItemTap
+        self.onItemAppear = onItemAppear
+        self._selectedThread = selectedThread
+    }
+
+    public var body: some View {
+        LazyVStack(spacing: 0) {
+            ForEach(threads) { thread in
+                factory.makeThreadListItem(
+                    thread: thread,
+                    threadDestination: threadDestination,
+                    selectedThread: $selectedThread
+                )
+                .background(factory.makeThreadListItemBackground(
+                    colors: colors,
```

Comment:
colors can be pulled in the view itself via the injection mechanism, no need to pass them here.

Reply:
I did it like this because the Channel List was working like this as well 🤔 So better not do it like this anymore?

Reply:
Done ✅